src/hotspot/share/interpreter/linkResolver.cpp

```diff
@@ -823,9 +823,8 @@ Method* LinkResolver::resolve_method(const LinkInfo& link_info,
   if (link_info.check_loader_constraints()) {
     // check loader constraints
     check_method_loader_constraints(link_info, resolved_method, "method", CHECK_NULL);
-
-    check_restricted_method(link_info, resolved_method, CHECK_NULL);
   }
+  check_restricted_method(link_info, resolved_method, CHECK_NULL);
```

Comment:
Can we please rename this as "check_native_access" to make terminology consistent